ldes-fragmentisers/ldes-fragmentisers-timebased-hierarchical/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebasedhierarchical/HierarchicalTimeBasedFragmentationStrategy.java

```diff
@@ -70,4 +75,33 @@ private Observation startFragmentationObservation(Observation parentObservation)
 				.start();
 	}
 
+	private Optional<LocalDateTime> getFragmentationObjectLocalDateTime(Model model, String subjectFilter,
+																			  String fragmentationPredicate) {
+		return getFragmentationObjectsLocalDateTime(model, subjectFilter, fragmentationPredicate)
+				.stream()
+				.findFirst();
+	}
+
+	private List<LocalDateTime> getFragmentationObjectsLocalDateTime(Model model, String subjectFilter,
+																		   String fragmentationPath) {
+		return model
+				.listStatements(null, ResourceFactory.createProperty(fragmentationPath), (Resource) null)
+				.toList()
+				.stream()
+				.filter(statement -> statement.getSubject().toString().matches(subjectFilter))
+				.map(this::getDateTimeValue)
+				.filter(Optional::isPresent)
+				.map(Optional::get)
+				.toList();
+	}
```

Comment:
I understand the reasoning for separating these methods, but doing a toList() on line 95 and then getting a stream on line 81 seems contraproductive